### ISSUE TITLE
Stop converting constructors universe instances (and parameters too)

### ISSUE THREAD

**Issue by mattam82** [coq/coq#6942]
#### Description of the problem

The invariants of conversion and unification should ensure that we only compare terms that have a common supertype. In this case it means for constructors we could stop converting their universe instances and parameters (they have indeed to be applied to the same number of parameters and by invariant they should be convertible). This could have an impact on unification, to be experimented.

**Comment by SkySkimmer** [comment]
I experimented with skipping conversion of parameters (and domain of lambdas), see https://github.com/coq/coq/wiki/Performance-experiments#skipping-conversion-of-parameters-of-constructors-and-of-the-domain-of-lambdas-failed
TLDR with the current right-to-left order of processing application arguments converting parameters/domains allows early failure so it doesn't work out.

**Comment by ppedrot** [comment]
Wasn't that solved by #6747?

**Comment by SkySkimmer** [comment]
No, that was only the universe instances.

**Comment by SkySkimmer** [comment]
Doesn't work out as a perf improvement in practice.
